internal/cloud/atlas/groups.go

```diff
@@ -17,27 +17,58 @@ type Group struct {
 	Name string `json:"name"`
 }
 
+type link struct {
+	Href string `json:"href"`
+	Rel  string `json:"rel"`
+}
+
 type groupResponse struct {
 	Results []Group `json:"results"`
+	Links   []*link `json:"links"`
 }
 
 func (c *client) Groups() ([]Group, error) {
-	res, resErr := c.do(
-		http.MethodGet,
-		groupsPath,
-		api.RequestOptions{},
-	)
-	if resErr != nil {
-		return nil, resErr
+	groupRes, err := c.fetchGroups(groupsPath, false)
+	if err != nil {
+		return nil, err
+	}
+
+	allGroups := groupRes.Results
+	next := groupRes.nextLink()
+	for next != nil {
```

Comment:
we may need to tweak this logic a bit, as _each_ groups response has the potential for a `"next"` link in it, not just that the initial response would have one

I think we can get a little sleek with it here:

```go
func (c *client) Groups() ([]Group, error) {
	var fetcher func(url string) ([]Group, error)
	fetcher = func(url string) ([]Group, error) {
		res, resErr := c.doURL(
			http.MethodGet,
			groupsPath,
			api.RequestOptions{},
		)
		if resErr != nil {
			return nil, resErr
		}
		if res.StatusCode != http.StatusOK {
			return nil, api.ErrUnexpectedStatusCode{"get groups", res.StatusCode}
		}
		defer res.Body.Close()

		var groupRes groupResponse
		if err := json.NewDecoder(res.Body).Decode(&groupRes); err != nil {
			return nil, err
		}

		allGroups := groupRes.Results
		for _, link := range groupRes.Links {
			if link.Rel == "next" {
				nextGroups, err := fetcher(link.Href)
				if err != nil {
					return nil, err
				}
				allGroups = append(allGroups, nextGroups...)
			}
		}
		return allGroups, nil
	}

	return fetcher(groupsPath)
}```

the main thing I'd tweak above there is it'd be nice if we could order the returned array in the same way we got the results (whereas I think how it is currently, we end up "reverse ordering" the pages)

Reply:
My understanding was that the group response (first & only group response) will contain the "links" field that is essentially a chain linking ****all** of the other groups** with the "next" field. 
https://www.mongodb.com/docs/atlas/api/atlas-admin-api/#std-label-atlas-api-linking
<img width="801" alt="image" src="https://user-images.githubusercontent.com/66972585/167275631-1588a34a-2f43-4ed3-918e-bc202cb1b8e3.png">
Does that sound right? or are you fairly certain that each group response could have its own chain link of projects?

Reply:
I like this! 
and in regards to the ordering, it looks like it is being kept in the proper order (since we are appending the new groups to the first). 
based off a walk through this is how the allGroups is being filled:
```
get group 1
get group 2 
get group 3
append group 3 to group 2
append groups [2,3] to group 1
```
allGroups is [1,2,3] ✅